public/pages/workflow_detail/workflow_inputs/workflow_inputs.tsx

```diff
@@ -188,73 +192,111 @@ export function WorkflowInputs(props: WorkflowInputsProps) {
-                  dispatch(getWorkflow(updatedWorkflow.id as string));
-                });
-              })
-              .catch((error: any) => {
-                console.error('Error provisioning updated workflow: ', error);
-              });
-          })
-          .catch((error: any) => {
-            console.error('Error updating workflow: ', error);
+    if (reprovision) {
+      await dispatch(
+        updateWorkflow({
+          workflowId: updatedWorkflow.id as string,
+          workflowTemplate: reduceToTemplate(updatedWorkflow),
+          reprovision: true,
+        })
+      )
+        .unwrap()
+        .then(async (result) => {
+          await sleep(1000);
+          success = true;
+          // Kicking off an async task to re-fetch the workflow details
+          // after some amount of time. Provisioning will finish in an indeterminate
+          // amount of time and may be long and expensive; we add this single
+          // auto-fetching to cover the majority of provisioning updates which
+          // are inexpensive and will finish within milliseconds.
+          new Promise((f) => setTimeout(f, 1000)).then(async () => {
+            dispatch(getWorkflow(updatedWorkflow.id as string));
           });
-      })
-      .catch((error: any) => {
-        console.error('Error deprovisioning workflow: ', error);
-      });
+        })
+        .catch((error: any) => {
+          console.error('Error reprovisioning workflow: ', error);
+        });
+    } else {
+      await dispatch(
+        deprovisionWorkflow({
+          workflowId: updatedWorkflow.id as string,
+          resourceIds: getResourcesToBeForceDeleted(props.workflow),
+        })
+      )
+        .unwrap()
+        .then(async (result) => {
+          await dispatch(
+            updateWorkflow({
+              workflowId: updatedWorkflow.id as string,
+              workflowTemplate: reduceToTemplate(updatedWorkflow),
+              reprovision: false,
+            })
+          )
+            .unwrap()
+            .then(async (result) => {
+              await sleep(1000);
+              await dispatch(provisionWorkflow(updatedWorkflow.id as string))
+                .unwrap()
+                .then(async (result) => {
+                  await sleep(1000);
+                  success = true;
+                  // Kicking off an async task to re-fetch the workflow details
+                  // after some amount of time. Provisioning will finish in an indeterminate
+                  // amount of time and may be long and expensive; we add this single
+                  // auto-fetching to cover the majority of provisioning updates which
+                  // are inexpensive and will finish within milliseconds.
+                  new Promise((f) => setTimeout(f, 1000)).then(async () => {
+                    dispatch(getWorkflow(updatedWorkflow.id as string));
+                  });
```

Comment:
```suggestion
                  setTimeout(async () => {
                     dispatch(getWorkflow(updatedWorkflow.id as string));
              }, 1000);
```

Reply:
nice. Will update manually in both places.